### DISCUSSION TITLE
Is there a way for a transaction to apply to every executor?

### DISCUSSION THREAD

**Discussion by prryplatypus** [ahopkins/mayim#35] [Q&A] [answered]
Hello,

I currently have to use 2 different executors within one single transaction, so I have 2 levels of "async with" in order to achieve this:
```py
thread_executor = self._get_thread_executor()
category_executor = self._get_category_executor()
async with thread_executor.transaction():
    async with category_executor.transaction():
        ...
```

I was wondering if there is something like a "global transaction" that can apply to every query used inside it. Something like this:
```py
thread_executor = self._get_thread_executor()
category_executor = self._get_category_executor()
async with Mayim.transaction():
    ...
```

**Comment by ahopkins** [comment, marked as answer]
No, but there should be. However, I am pretty sure that under the hood `psycopg` would handle that with savepoints.
